x/gov/keeper/grpc_query.go

```diff
@@ -1,7 +1,183 @@
 package keeper
 
 import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/maticnetwork/heimdall/x/gov/types"
 )
 
 var _ types.QueryServer = Keeper{}
+
+// Proposal returns proposal details based on ProposalID
+func (q Keeper) Proposal(c context.Context, req *types.QueryProposalRequest) (*types.QueryProposalResponse, error) {
```

Comment:
I suggest using k instead of q for naming consistency. @anilCSE is there any specific reason why q used in cosmos-sdk ?